src/partial-assembly.md

```diff
@@ -0,0 +1,171 @@
+structure of the degrees of freedom and quadrature points on quadrilateral and
+hexahedral elements to perform the action of **B** without storing it as a
+matrix.
+
+The partial assembly algorithm requires the optimal amount of memory transfers
+(with respect to the polynomial order) and near-optimal FLOPs for operator
+evaluation. It consists of an operator *setup* phase, that evaluates and stores
+**D** and an operator *apply* (evaluation) phase that computes the action of
+**A** on an input vector. When desired, the setup phase may be done as a
+side-effect of evaluating a different operator, such as a nonlinear residual.
+The relative costs of the setup and apply phases are different depending on the
+physics being expressed and the representation of **D**.
+
+### Parallel Decomposition
+
+After the application of each of the first three transition operators, **P**,
+**G** and **B**, the operator evaluation is decoupled on their ranges, so
+**P**, **G** and **B** allow us to "zoom-in" to subdomain, element and
+quadrature point level, ignoring the coupling at higher levels.
+
+Thus, a natural mapping of **A** on a parallel computer is to split the
+**T-vector** over MPI ranks (a non-overlapping decomposition, as is typically
+used for sparse matrices), and then split the rest of the vector types over
+computational devices (CPUs, GPUs, etc.) as indicated by the shaded regions in
+the diagram above.
+
+One of the advantages of the decomposition perspective in these settings is
+that the operators **P**, **G**, **B** and **D** clearly separate the MPI
+parallelism in the operator (**P**) from the unstructured mesh topology
+(**G**), the choice of the finite element space/basis (**B**) and the geometry
+and point-wise physics **D**. These components also naturally fall in different
+classes of numerical algorithms: parallel (multi-device) linear algebra for
+**P**, sparse (on-device) linear algebra for **G**, dense/structured linear
+algebra (tensor contractions) for **B** and parallel point-wise evaluations for
+**D**.
+
+## Partial Assembly for Discontinuous Galerkin methods
+
+A complementary partial assembly decomposition is used for Discontinuous
+Galerkin methods to handle face terms, where a similar sequence of operators is
+applied on the faces to compute the numerical fluxes. However, since elements
+are decoupled, the element restriction **G** is the identity, and a face
+restriction **G<sub>F</sub>** is used instead to compute the numerical fluxes
+and couple elements together. This face restriction **G<sub>F</sub>** goes from
+element degrees of freedom to face degrees of freedom. Then a **B<sub>F</sub>**
+operator can be applied on the faces. An analogous **D<sub>F</sub>** operator
+is then applied at the face quadrature points.
+
+Currently, we support partial assembly only for Gauss-Lobatto and Bernstein
+bases, with integrators that don't require derivatives on the faces.
```

Comment:
To be a bit more precise: "derivatives" -> "normal derivatives".

Reply:
Well, to be completely accurate, we don't support tangential derivatives, but we could support them easily, contrary to normal derivatives that require a bit of work.

Reply:
Yes, you are right. I thought you added the tangential derivatives. Anyway, the code for computing them is already there, in `FaceQuadratureInterpolator`, so it is just a matter of storing them if they are requested. The normal derivatives, on the other hand, will require different face-restriction and different face-quadrature-interpolator.